File: src/commands/SettingsCommands.ts
import { invoke } from '@tauri-apps/api/core';
import type { AppSettings } from './types/settings.types';

/**
 * 设置管理命令
 */
export class SettingsCommands {

  /**
   * 保存系统托盘状态
   * @param enabled 是否启用
   * @returns 保存后的状态
   */
  static async saveSystemTrayState(enabled: boolean): Promise<boolean> {
    return invoke('save_system_tray_state', { enabled });
  }

  /**
   * 保存静默启动状态
   * @param enabled 是否启用
   * @returns 保存结果消息
   */
  static async saveSilentStartState(enabled: boolean): Promise<boolean> {
    return invoke('save_silent_start_state', { enabled });
  }

  /**
   * 保存隐私模式状态
   * @param enabled 是否启用
   * @returns 保存后的状态
   */
  static async savePrivateModeState(enabled: boolean): Promise<boolean> {
    return invoke('save_private_mode_state', { enabled });
  }

  /**
   * 保存 Debug Mode 状态
   * @param enabled 是否启用
   * @returns 保存后的状态
   */
  static async saveDebugModeState(enabled: boolean): Promise<boolean> {
    return invoke('save_debug_mode_state', { enabled });
  }

  /**
   * 获取所有应用设置
   * @returns 应用设置对象
   */
  static async getAll(): Promise<AppSettings> {
    return invoke('get_all_settings');
  }
}

File: src-tauri/src/commands/settings_commands.rs
//! Application settings commands
//! Responsible for managing and storing application configuration, using State pattern

use tauri::{AppHandle, Manager};

/// Save system tray state
#[tauri::command]
pub async fn save_system_tray_state(app: AppHandle, enabled: bool) -> Result<bool, String> {
    crate::log_async_command!("save_system_tray_state", async {
        let system_tray = app.state::<crate::system_tray::SystemTrayManager>();

        if enabled {
            system_tray.enable(&app)?;
        } else {
            system_tray.disable(&app)?;
        }

        let settings_manager = app.state::<crate::app_settings::AppSettingsManager>();
        let settings = settings_manager.get_settings();
        Ok(settings.system_tray_enabled)
    })
}

/// Save silent start state
#[tauri::command]
pub async fn save_silent_start_state(app: AppHandle, enabled: bool) -> Result<bool, String> {
    crate::log_async_command!("save_silent_start_state", async {
        let settings_manager = app.state::<crate::app_settings::AppSettingsManager>();

        settings_manager.update_settings(|settings| {
            settings.silent_start_enabled = enabled;
        })?;

        let settings = settings_manager.get_settings();
        Ok(settings.silent_start_enabled)
    })
}

/// Save privacy mode state
#[tauri::command]
pub async fn save_private_mode_state(app: AppHandle, enabled: bool) -> Result<bool, String> {
    crate::log_async_command!("save_private_mode_state", async {
        let settings_manager = app.state::<crate::app_settings::AppSettingsManager>();

        settings_manager.update_settings(|settings| {
            settings.private_mode = enabled;
        })?;

        let settings = settings_manager.get_settings();
        Ok(settings.private_mode)
    })
}

/// Save Debug Mode state
#[tauri::command]
pub async fn save_debug_mode_state(app: AppHandle, enabled: bool) -> Result<bool, String> {
    crate::log_async_command!("save_debug_mode_state", async {
        let settings_manager = app.state::<crate::app_settings::AppSettingsManager>();

        settings_manager.update_settings(|settings| {
            settings.debug_mode = enabled;
        })?;

        let settings = settings_manager.get_settings();
        Ok(settings.debug_mode)
    })
}

/// Get all application settings
#[tauri::command]
pub async fn get_all_settings(app: AppHandle) -> Result<serde_json::Value, String> {
    crate::log_async_command!("get_all_settings", async {
        let settings_manager = app.state::<crate::app_settings::AppSettingsManager>();
        let settings = settings_manager.get_settings();

        Ok(serde_json::json!({
            "system_tray_enabled": settings.system_tray_enabled,
            "silent_start_enabled": settings.silent_start_enabled,
            "debugMode": settings.debug_mode,
            "privateMode": settings.private_mode
        }))
    })
}
